doc/examples/edges/plot_canny.py

```diff
@@ -30,25 +31,70 @@
 im = ndi.gaussian_filter(im, 4)
 im += 0.2 * np.random.random(im.shape)
 
+# Generate lower SNR noisy image of a square
+im2 = np.zeros((128, 128))
+im2[32:-32, 32:-32] = 1
+
+im2 = ndi.rotate(im2, 15, mode='constant')
+im2 = ndi.gaussian_filter(im2, 4)
+im2 += 0.4 * np.random.randn(*(im2.shape))
+
 # Compute the Canny filter for two values of sigma
-edges1 = feature.canny(im)
-edges2 = feature.canny(im, sigma=3)
+edges1_1 = feature.canny(im)
+edges1_2 = feature.canny(im, sigma=3)
+edges1_3 = feature.canny(im, use_quantiles=True, sigma=3)
+edges1_4 = feature.canny(im, sigma=3, high_threshold=threshold_otsu,
+                         low_threshold=threshold_otsu)
+
+edges2_1 = feature.canny(im2)
+edges2_2 = feature.canny(im2, sigma=3)
+edges2_3 = feature.canny(im2, sigma=3, use_quantiles=True)
+edges2_4 = feature.canny(im2, sigma=3, low_threshold=threshold_otsu,
+                         high_threshold=threshold_otsu)
 
 # display results
-fig, (ax1, ax2, ax3) = plt.subplots(nrows=1, ncols=3, figsize=(8, 3),
+fig, axes = plt.subplots(nrows=2, ncols=5, figsize=(12, 6),
                                     sharex=True, sharey=True)
 
-ax1.imshow(im, cmap=plt.cm.gray)
-ax1.axis('off')
-ax1.set_title('noisy image', fontsize=20)
+axes[0][0].imshow(im, cmap=plt.cm.gray)
+axes[0][0].axis('off')
+axes[0][0].set_title('Low noise image')
```

Comment:
Can you change this to NumPy indexing? [0, 0], etc. Much neater imho. 

I also feel like some of this repetition can be removed: instead of individual calls to `axis('off')`, remove them all and at the end say, `for ax in axes.ravel(): ax.axis('off')`.